SalsaCompiler/SalsaCompiler/main.swift

```diff
@@ -59,23 +59,30 @@ let workingDirectory: String = {
   }
 }()
 
+func cleanupWorkingDirectory() {
+  try? FileManager.default.removeItem(atPath: workingDirectory)
+}
+
 // Create the working directory
 do {
   try FileManager.default.createDirectory(atPath: workingDirectory, withIntermediateDirectories: true, attributes: nil)
 } catch {
+  cleanupWorkingDirectory()
   fatalError("Failed to create directory: \(workingDirectory)")
```

Comment:
Since we're doing this all over the place, I think it'd feel a little nicer to extract the two into some sort of `salsaAbort`/`salsaFatalError` helper function that does the two. Then if we ever need more cleanup, it's also easier to add.

Bummer that we can't use defer for this...